src/Symfony/Component/Asset/EventListener/PreloadListener.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\EventListener;
+
+use Symfony\Component\Asset\Preload\PreloadManager;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Adds the preload Link HTTP header to the response.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class PreloadListener implements EventSubscriberInterface
+{
+    private $preloadManager;
+
+    public function __construct(PreloadManager $preloadManager)
+    {
+        $this->preloadManager = $preloadManager;
+    }
+
+    public function onKernelResponse(FilterResponseEvent $event)
+    {
+        if ($value = $this->preloadManager->getLinkValue()) {
```

Comment:
this logic must be done only for the master request.
An asset referenced in a subrequest must be preloaded by the master request, as it is the one being sent to the user, and your code would also use all previous master preloaded asset for the subrequest, thus breaking preloading.